sci-libs/vtk/vtk-9.0.1.ebuild

```diff
@@ -294,7 +294,25 @@ src_configure() {
 	fi
 
 	if use cuda; then
-		mycmakeargs+=( -DVTKm_CUDA_Architecture="native" )
+		local MY_CUDA_ARCH=native
+		if [[ -n ${VTK_CUDA_ARCH} ]]; then
+			# we ignore fermi arch, because current nvidia-cuda-toolkit-11*
+			# no longer supports it
+			case ${VTK_CUDA_ARCH} in
```

Comment:
What I had in mind for default assignment was to potentially use it here, i.e. remove "if" entirely, and then add native to the list of valid values. Not that it changes much beside less indenting, I'm okay with it either way (better than before).

Reply:
I strongly disagree with your last statement. It was way better before. Easier to read, easier to follow the code flow, the structure more clean.

Reply:
I don't really get why a simple sanity check need to be complicated, but okay.

Reply:
I was also wondering if this check is really useful, i.e. could just pass the value if the build system knows to error out on unknown values.

Reply:
Build system doesn't fail during configuration, it just accepts invalid values. I just tried with `VTK_CUDA_ARCH=-DDEBUG ebuild vtk-9.0.1 clean configure`. The src_configure phase succeeds, the relevant values are
```
artus # grep VTKm_CUDA /var/tmp/portage/sci-libs/vtk-9.0.1/work/vtk-9.0.1_build/CMakeCache.txt 
VTKm_CUDA_Architecture:STRING=-DDEBUG
//STRINGS property for variable: VTKm_CUDA_Architecture
VTKm_CUDA_Architecture-STRINGS:INTERNAL=native;fermi;kepler;maxwell;pascal;volta;turing;all;none
```
It will probably fail during src_compile phase.

Reply:
Given it could take a while to fail, might as well keep the check then -- better off having `die` do it.